generators/reference_chunking/newline_splitter/code_snippet_common.md

```diff
@@ -0,0 +1,25 @@
+```python
+from typing import List 
+
+def newline_splitter(text: str) -> List[str]:
+    """
+    @param text: The input string that needs to be split.
+    @return:  A list of strings where each string is a non-empty line from the input.
+    """
+    splits = [t.strip() for t in text.split("\n")]
+    return [val for val in splits if len(val) > 0]
+
+# ↑ necessary bricks function 
+# -----------------------------------------------------------------------------------------
+# ↓ example implementation
+
+def example_integration():
+    texts = ["""
+    This is a sentences.
+    This too, but in another line
+    """, "This is a sentence\nwith a newline literal!"]
+    for text in texts:
+        print(f"The text {text} was split into {newline_splitter(text)}")
```

Comment:
<img width="627" alt="Screenshot 2023-10-17 at 13 18 47" src="https://github.com/code-kern-ai/bricks/assets/125893779/bec50a5d-772d-49e6-83fe-6172aa5d0226">


The output looks a little weird, and I think in that format it's harder to understand what's happening.
Could use `{repr(text)}` instead.

Reply:
That's a great idea. I've added it!